File: leetcode/c/isAnagram.c
class Solution {
public:
    bool isAnagram(string s, string t) {
        char tmp[256]= {0};
        for (int i =0;i <s.length();i++){
            tmp[s[i]]++;
        }
        for(int i=0;i<t.length();i++){
            tmp[t[i]]--;
        }
        for(int i=0;i<256;i++){
            if(0 != tmp[i]) return false;
        }
        return true;
    }
};

File: leetcode/python/isAnagram.py


class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # 68 ms
        # return len(s) == len(t) and sorted(s) == sorted(t)
        # 76 ms
        if len(s) != len(t):
            return False
        dicts = collections.defaultdict(int)
        for i in range(len(s)):
            dicts[s[i]] = dicts[s[i]] + 1
            dicts[t[i]] = dicts[t[i]] - 1
        for val in dicts.values():
            if val != 0:
                return False
        return True
